File: python/leetcode/311.py
class Solution:
    def multiply(self, A, B):
        m, n, l = len(A), len(A[0]), len(B[0])
        C = [[0 for i in range(l)] for j in range(m)]
        for i in range(m):
            for j in range(n):
                if A[i][j] != 0:
                    for k in range(l):
                        if B[j][k]] != 0: 
                            C[i][k] += A[i][j] * B[j][k]
        return C

File: cpp/leetcode/311.cpp
class Solution {
public:
    vector<vector<int>> multiply(vector<vector<int>>& A, vector<vector<int>>& B) {
        int m = A.size(), n = B.size(), l = B[0].size();
        vector<vector<int>> C(m, vector<int>(l, 0));
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                if (A[i][j])
                    for (int k = 0; k < l; k++)
                      if(B[j][k]) C[i][k] += A[i][j] * B[j][k];
        return C;
    }
};
